### PR TITLE
fix: do not create duplicate auto PRs

### LINKED ISSUE #1213: change how automatic PRs are handled
Right now the automatic PRs are created when there is a push to a branch matching `auto-update/**` - this means that we get duplicate PRs when the branches are updated because it counts as a push. 

Two options:

1. the PR could be made by the component pushing the branch via a github API call
2. the auto-PR action can stay in this repo, but does not run if a PR for that branch is already running

### PR DESCRIPTION
**Pull request by RenkuBot**
closes #1213

### CODE REVIEW

File: .github/workflows/auto-reqs-update-pr.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
           [[ ${GITHUB_REF} =~ $regex ]]
           pr_repo=$BASH_REMATCH
           # find other PRs from that repo
-          other_prs=$(curl -s ${renku_repo}/pulls | jq ".[] | select(.head.ref | test(\"auto-update/${pr_repo}\")) | .number")
```
Comment (ableuler) [inline .github/workflows/auto-reqs-update-pr.yml:18]: hmm sorry, don't get it yet: `.head.ref` will be `auto-update/renku-testing-1234`, but won't GITHUB_REF be something like `refs/pull/1493` ?

Comment (rokroskar) [reply .github/workflows/auto-reqs-update-pr.yml:18]: don't think so - the event is a branch push not a PR update.

File: .github/workflows/auto-reqs-update-pr.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
...
+          other_prs=$(curl -s ${renku_repo}/pulls | jq ".[] | select((.head.ref | test(\"auto-update/${pr_repo}\")) and (.head.ref != \"${GITHUB_REF:11}\")) | .number")
```
Comment (ableuler) [inline .github/workflows/auto-reqs-update-pr.yml:18]:
```suggestion
          other_prs=$(curl -s ${renku_repo}/pulls | jq ".[] | select((.head.ref | test(\"auto-update/${pr_repo}\")) and (.head.ref != \"${GITHUB_REF:12}\")) | .number")
```
I think you also have to remove the slash.

File: .github/workflows/auto-reqs-update-pr.yml (hunk 1 above)
Comment (ableuler) [reply .github/workflows/auto-reqs-update-pr.yml:18]: ok, get it!

**Review by ableuler — CHANGES_REQUESTED** [review]

File: .github/workflows/auto-reqs-update-pr.yml (hunk 2 above)
Comment (rokroskar) [reply .github/workflows/auto-reqs-update-pr.yml:18]: ah thanks, in the test string I had I accidentally removed the first slash...

**Comment by rokroskar** [comment]
obviously it doesn't work

image

😢

File: .github/workflows/auto-reqs-update-pr.yml (hunk 2 above)
Comment (rokroskar) [reply .github/workflows/auto-reqs-update-pr.yml:18]: turns out I had it right :) the counting starts at zero

```
> echo $REF
refs/heads/auto-update/renku-testing-1234
>  echo ${REF:11}
```

Comment (ableuler) [reply .github/workflows/auto-reqs-update-pr.yml:18]: aaa, sorry - I thought you wanted to cut the `auto-update/`, from `auto-update/renku-testing-1234` which wouldn't make much sense. Sorry, my bad.

**Review by ableuler — APPROVED** [review]

File: .github/workflows/auto-reqs-update-pr.yml (hunk 2 above)
Comment (rokroskar) [reply .github/workflows/auto-reqs-update-pr.yml:18]: no worries, I'm usually off by one on these sorts of things :)

**Pull request merged**